### DISCUSSION TITLE
Usage of StateMemoryStorage()

### DISCUSSION THREAD

**Discussion by fadelazzahraa** [eternnoir/pyTelegramBotAPI#1570] [Q&A]
### Usage of StateMemoryStorage()
I saw an example of `StateMemoryStorage` in [custom_states.py](https://github.com/eternnoir/pyTelegramBotAPI/blob/master/examples/custom_states.py) which is to store what the specific user wrote in bot.

My question is: Can I use `StateMemoryStorage` without using `set_state`? 
In this case, I don't need a 'state' marker for what stages the user has done, I just want to store what specific user wrote in bot (can't be overwritten by another user)

or maybe someone can share another implementation of StateMemoryStorage. Thanks

**Comment by coder2020official** [comment]
Hello. Sorry for a loong response.
Btw, you can store data there... But the purpose of this class was for states..

**Reply by fadelazzahraa** [reply]
Are there more efficient method to store data from specific user rather than using StateMemoryStorage()?

**Reply by sudousercom** [reply]
mysql, redis or mongodb?
